common_features.mk

```diff
@@ -28,12 +28,43 @@ ifeq ($(strip $(API_SYSEX_ENABLE)), yes)
     SRC += $(QUANTUM_DIR)/api.c
 endif
 
+## Audio feature
+#TODO: some keyboards set ENABLE=yes, but did not configure any audio pins having effectively a non-functioning audio-system bloat the firmware; remove this block as soon as each driver enforces a correct pin configuration
+ifeq ($(strip $(AUDIO_PIN))$(strip $(AUDIO_PIN_ALT)),)
+AUDIO_ENABLE := no
+else
+AUDIO_ENABLE ?= no
+endif
 ifeq ($(strip $(AUDIO_ENABLE)), yes)
+    ifeq ($(PLATFORM),CHIBIOS)
+        AUDIO_DRIVER ?= dac_basic
+        ifeq ($(strip $(AUDIO_DRIVER)), dac_basic)
+            OPT_DEFS += -DAUDIO_DRIVER_DAC
+        else ifeq ($(strip $(AUDIO_DRIVER)), dac_additive)
+            OPT_DEFS += -DAUDIO_DRIVER_DAC
+        ## stm32f2 and above have a usable DAC unit, f1 do not, and need to use pwm instead
+        else ifeq ($(strip $(AUDIO_DRIVER)), pwm_software)
+            OPT_DEFS += -DAUDIO_DRIVER_PWM
+        else ifeq ($(strip $(AUDIO_DRIVER)), pwm_hardware)
+            OPT_DEFS += -DAUDIO_DRIVER_PWM
+        endif
+    else
+        # fallback for all other platforms is pwm
+        AUDIO_DRIVER ?= pwm_hardware
+        OPT_DEFS += -DAUDIO_DRIVER_PWM
+    endif
+    ifdef AUDIO_PIN
```

Comment:
don't quite follow... all that is done here is the selection of an audio-driver and audio-pin(s); the finetuning/configuration is still done in `config.h` (like dac sample rate, ...)
the nice thing about doing the selection here, is that it can be easily transformed into platformspecific defines, which then enable certain codpaths (e.g. special AVR pins: AUDIO_PIN=C5 -> enable pwm for the pwm-capable C5 pin, which are enclosed in an ifdef AUDIO_PIN_C5)

Reply:
> in short: doing the pin assignment in the makefile is more flexible/consistent since its a step prior to the gcc-preprocessor

Except its inconsistent with the other features and the direction we want to go right now. 
Driver and enable/disable are set in rules.mk, the other config options go in config.h. The inconsistencies in naming is purely something that can be solved if required by renaming things within driver implementations and maybe adding some backward compatibility.

Reply:
ok, changed it with 43e5032 - this what you had in mind?